internal/cutil/ptrguard.go

```diff
@@ -1,90 +1,37 @@
+//go:build !go1.21
```

Comment:
This is more of a learning for me / double check, can you confirm if this will work correctly on go 1.22 and later? I searched for the precise syntax here and only found articles on the general syntax of build tags and the change from "+build" to "go:build". *sigh*

Reply:
"a term for each Go major release, through the current version: "go1.1" from Go version 1.1 onward, "go1.12" from Go 1.12, and so on."
https://pkg.go.dev/cmd/go#hdr-Build_constraints